server/src/main/java/hu/flowacademy/epsilon/sport_event_organizer/service/CupService.java

```diff
@@ -54,30 +73,23 @@ public Cup getByName(String name) {
         throw new CupNotFoundException(name);
     }
 
+    // returning all non-deleted cups!
     public List<Cup> getAllCups() {
-        return cupRepository.findAll();
-    }
-
-    public List<Cup> getAllNonDeletedCups() {
         List<Cup> cupList = cupRepository.findAll();
-        for (Cup cup: cupList) {
-            if (cup.isDeleted()) {
-                cupList.remove(cup);
-            }
-        }
-        log.info(cupList.toString());
-        return cupList;
+        return cupList.stream()
+                .filter(cup -> !cup.isDeleted())
+                .collect(Collectors.toList());
     }
 
-    // returning only non-deleted cups!
+    // returning only non-deleted cups - by place!
     public List<Cup> getCupsByPlace(String place) {
         List<Cup> cupList = cupRepository.findByPlace(place);
         return cupList.stream()
                 .filter(cup -> !cup.isDeleted())
```

Comment:
Használjátok inkább ezt: `Predicate.not(Cup::isDeleted)`

Reply:
innen: [link](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/function/Predicate.html#not(java.util.function.Predicate))